yt_dlp/downloader/hls.py

```diff
@@ -150,12 +157,12 @@ def is_ad_fragment_end(s):
         i = 0
         media_sequence = 0
         decrypt_info = {'METHOD': 'NONE'}
-        external_aes_key = info_dict.get('hls_aes_key')
-        if isinstance(external_aes_key, str):
+        external_aes_key = traverse_obj(info_dict, ('hls_aes', 'key', {str}))
```

Comment:
```suggestion
        external_aes_key = traverse_obj(info_dict, ('hls_aes', 'key'))
```
If the extractor sets non string value, it should error

Same below